engineering-decision-records/007-observability.md

```diff
@@ -0,0 +1,52 @@
+---
+type: architecture
+status: accepted
+---
+
+# Observability
+
+## Context
+
+This decision record documents a general organization for how the Amaru node will collect and report critical operational metrics, logs, and tracing data.
+
+## Motivation
+
+A key component of operating any computing system at scale is observability: tracking what happened, how often something happened, how long it took, how much resources something is consuming, and where potential issues or bottlenecks might be found.
+
+Thus, we would like Amaru to track and report these traces and metrics. We would also like to use industry standards so that users of the Amaru node can swap out their own visualization and aggregation infrastructure as needed.
+
+We would like the codebase to organize the metrics it tracks in a simple, consistent and modular way, such that each module or crate can own a subset of the metrics, while at the same time being consistent and discoverable through reading the code.
+
+## Decision
+
+We will use the [OpenTelemetry](https://opentelemetry.io/) standard for reporting metrics, and provide at least two example setups for collecting and monitoring these metrics:
+- [Jaeger](https://www.jaegertracing.io/)
+- [Grafana](https://grafana.com/) and [Tempo](https://grafana.com/oss/tempo/)
+
+We will use the [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) crate to collect and report traces and metrics.
+
+We will make judicious use of [Spans](https://opentelemetry.io/docs/concepts/observability-primer/#spans) to expose the structured nature of the workload the Amaru node performs.
+
+Each top level module or crate will (optionally) define its own Metrics module, which exposes a `{CrateName}Metrics` struct, with a `new(..)` method and deriving `Clone`. At the time of this decision record, this would be the Amaru binary, Conesnsus, the Ledger, and the Sync module.
```

Comment:
```suggestion
Each top level module or crate will (optionally) define its own Metrics module, which exposes a `{CrateName}Metrics` struct, with a `new(..)` method and deriving `Clone`. At the time of this decision record, this would be the Amaru binary, Consensus, the Ledger, and the Sync module.
```